### ISSUE TITLE
SPARQL-Star grammar imprecisions

### ISSUE THREAD

**Issue by enizor** [w3c/rdf-star#256]
With the proposed SPARQL-Star grammar :

```EBNF
QuotedTP 		::= 	'<<' qtSubjectOrObject Verb qtSubjectOrObject '>>'
QuotedTriple 		::= 	'<<' DataValueTerm ( iri | 'a' ) DataValueTerm '>>'
qtSubjectOrObject 	::= 	Var | BlankNode | iri | RDFLiteral | NumericLiteral | BooleanLiteral | QuotedTP
DataValueTerm 		::= 	iri | RDFLiteral | NumericLiteral | BooleanLiteral | QuotedTriple
```
the subject of any quoted triple is allowed to be a literal, and the `DataValueTerm` rule (used for VALUES clauses) prevents the subject/object of the quoted triple to be a blank node.

It seems necessary to create different rules for quoted subjects and object, such as:

```EBNF
QuotedTP 		::= 	'<<' qtSubjectOrObject Verb qtSubjectOrObject '>>'
QuotedTriple 		::= 	'<<' DataValueTerm ( iri | 'a' ) DataValueTerm '>>'
qtSubject 		::= 	Var | BlankNode | iri | QuotedTP
qtObject 		::= 	Var | BlankNode | iri | RDFLiteral | NumericLiteral | BooleanLiteral | QuotedTP
DataValueTermSubject 	::= 	iri | BlankNode | QuotedTriple
DataValueTermObject 	::= 	iri | BlankNode | RDFLiteral | NumericLiteral | BooleanLiteral | QuotedTriple
```

**Comment by afs** [comment]
> the subject of any quoted triple is allowed to be a literal

Yes - SPARQL allows literals in the subject position.  `{"foo":q "bar"}` is legal syntax and the spec noted it will never match data. It arises naturally in the definition of SPARQL.

(See also "generalized RDF" in RDF 1.1). 

> prevents the subject/object of the quoted triple to be a blank node.

`QuotedTP` is "quoted triple pattern" and allows variables and blank nodes that match during BGP matching.
`QuotedTriple` is a constant term (no variables).

Blank nodes in patterns behave like variables - they match the data. VALUES does not involved matching and so not allow blank node syntax. So in the same way QuotedTriple variables or blank nodes.

**Comment by pchampin** [comment]
> `{"foo":q "bar"}` is legal syntax and the spec noted it will never match data.

Is it absolutely correct, though? Under some inference regimes, triples with literals as subjects can be inferred. E.g.:
```
:answer owl:sameAs 42 .
```
would entail
```
42 owl:sameAs :answer .
```
And so
```
SELECT * { 42 ?p ?o }
```
should match if that entailment regime was supported, right?

**Comment by hartig** [comment]
owl:sameAs is defined only for individuals, not for literals (that is, according to the [RDF-based semantics of OWL](http://www.w3.org/TR/owl2-rdf-based-semantics/), the rdfs:domain of owl:sameAs is owl:Thing; and a literal is not an owl:Thing). Therefore, writing
```
_answer owl:sameAs 42 .
```
should already be inconsistent in itself. No?

**Comment by enizor** [comment]
My mistake, sorry for the noise.

**Comment by afs** [comment]
"we"?

**Comment by pchampin** [comment]
> My mistake, sorry for the noise.

I interpret this issue as closed. If I'm wrong, feel free to reopen it on w3c/sparql-query as the WG has taken on this work.
